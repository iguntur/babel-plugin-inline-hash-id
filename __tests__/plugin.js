'use strict';
const pluginTester = require('babel-plugin-tester');
const plugin = require('../lib/hash-id');

pluginTester({
	plugin,
	pluginName: 'inline-hash-id',
	babelOptions: {
		presets: [
			'react'
		]
	},
	tests: [
		{
			title: 'basic generate hash',
			code: `__hashId('ðŸŒˆ')`,
			snapshot: true
		},
		{
			title: 'with option',
			code: `__hashId('ðŸŒˆ', {uniqPerFile: false})`,
			snapshot: true
		},
		{
			title: 'with react #1',
			code: `
				React.createElement('div', {
					id: __hashId('root')
				});

				React.createElement('div', {
					id: __hashId('root', {uniqPerFile: false})
				});
			`,
			snapshot: true
		},
		{
			title: 'with react #2',
			code: `
				<body>
					<div id={__hashId('root')}></div>
				</body>
			`,
			snapshot: true
		},
		{
			title: 'inside function',
			code: `
				document.getElementById(__hashId('root'));
			`,
			snapshot: true
		}
	]
});
