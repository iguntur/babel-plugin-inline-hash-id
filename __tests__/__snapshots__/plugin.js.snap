// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inline-hash-id basic generate hash: basic generate hash 1`] = `
"
__hashId('ðŸŒˆ')

      â†“ â†“ â†“ â†“ â†“ â†“

'yrcJgmjiVQaO';
"
`;

exports[`inline-hash-id inside function: inside function 1`] = `
"
document.getElementById(__hashId('root'));

      â†“ â†“ â†“ â†“ â†“ â†“

document.getElementById('qsmageWVVzIP');
"
`;

exports[`inline-hash-id with option: with option 1`] = `
"
__hashId('ðŸŒˆ', {uniqPerFile: false})

      â†“ â†“ â†“ â†“ â†“ â†“

'fZRbeFUKoFBH';
"
`;

exports[`inline-hash-id with react #1: with react #1 1`] = `
"
React.createElement('div', {
	id: __hashId('root')
});

React.createElement('div', {
	id: __hashId('root', {uniqPerFile: false})
});

      â†“ â†“ â†“ â†“ â†“ â†“

React.createElement('div', {
	id: 'qsmageWVVzIP'
});

React.createElement('div', {
	id: 'SBNJTRNFjGow'
});
"
`;

exports[`inline-hash-id with react #2: with react #2 1`] = `
"
<body>
	<div id={__hashId('root')}></div>
</body>

      â†“ â†“ â†“ â†“ â†“ â†“

React.createElement(
	'body',
	null,
	React.createElement('div', { id: 'qsmageWVVzIP' })
);
"
`;
